interceptors/client.go

```diff
@@ -74,7 +75,10 @@ func (s *monitoredClientStream) RecvMsg(m interface{}) error {
 	if err == nil {
 		return nil
 	}
-
-	s.reporter.PostCall(err, time.Since(s.startTime))
+	var postErr error
+	if err != io.EOF {
+		postErr = err
+	}
+	s.reporter.PostCall(postErr, time.Since(s.startTime))
```

Comment:
The problem with passing `io.EOF` is that it's not a valid gRPC status code. The code there then turns it into status `Unknown` rather than `OK`, which is incorrect. A test was failing because of that.

Reply:
`io.EOF` is returned from `RecvMsg` to indicate the end of the stream (https://pkg.go.dev/google.golang.org/grpc#ClientStream), does this no break that behaviour?

Reply:
No it does not - note that the function still returns `err` i.e. `io.EOF`, we only change what we pass to `PostCall()`.

Reply:
Fair enough, makes sense.